routes/weatherProviders/DarkSky.ts

```diff
@@ -5,37 +5,71 @@ import { httpJSONRequest } from "../weather";
 
 async function getDarkSkyWateringData( coordinates: GeoCoordinates ): Promise< WateringData > {
 	// The Unix timestamp of 24 hours ago.
-	const timestamp: number = moment().subtract( 1, "day" ).unix();
+	const yesterdayTimestamp: number = moment().subtract( 1, "day" ).unix();
+	const todayTimestamp: number = moment().unix();
 
 	const DARKSKY_API_KEY = process.env.DARKSKY_API_KEY,
-		historicUrl = `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${coordinates[0]},${coordinates[1]},${timestamp}`;
+		yesterdayUrl = `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${coordinates[0]},${coordinates[1]},${yesterdayTimestamp}?exclude=currently,minutely,daily,alerts,flags`,
+		todayUrl = `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${coordinates[0]},${coordinates[1]},${todayTimestamp}?exclude=currently,minutely,daily,alerts,flags`;
 
-	let historicData;
+	let yesterdayData, todayData;
 	try {
-		historicData = await httpJSONRequest( historicUrl );
+		yesterdayData = await httpJSONRequest( yesterdayUrl );
+		todayData = await httpJSONRequest( todayUrl );
 	} catch (err) {
 		// Indicate watering data could not be retrieved if an API error occurs.
 		return undefined;
 	}
 
+	if ( !todayData.hourly || !todayData.hourly.data || !yesterdayData.hourly || !yesterdayData.hourly.data ) {
+		return undefined;
+	}
+
+	// Fail if not enough data is available.
+	if ( yesterdayData.hourly.data.length + todayData.hourly.data.length < 24 ) {
+		return undefined;
+	}
+
+	const totals = { temp: 0, humidity: 0, precip: 0 };
+	// The number of hourly forecasts from today's data that use historic data (not forecast data).
+	// Find the first element that contains forecast data.
+	const todayHistoricElements = todayData.hourly.data.findIndex( ( data ) => data.time > todayTimestamp - 60 * 60 );
+	// Sum data from the current calendar day.
+	const todayPeriods =  Math.min( 24, todayHistoricElements );
+	for ( let index = todayPeriods - 1; index >= 0; index-- ) {
+		totals.temp += todayData.hourly.data[ index ].temperature;
+		totals.humidity += todayData.hourly.data[ index ].humidity;
+		totals.precip += todayData.hourly.data[ index ].precipIntensity
+	}
+
+	// Sum data from yesterday.
+	for ( let index = 24 - todayPeriods - 1; index >= 0; index-- ) {
```

Comment:
Great idea, I was wondering how we could make that code a bit cleaner. I agree that cleaner code is more important than shaving a few microseconds off the runtime, so I'll update based on your suggestion. 
I think the rest of the suggestions are out of this PR's scope and are something that @salbahra should check out since he's the effective product owner for the app, and any breaking changes should go through him.

Reply:
Remaining to be resolving through Issue #47 .